keras_nlp/layers/modeling/rotary_embedding.py

```diff
@@ -115,8 +117,12 @@ def _compute_cos_sin_embedding(self, inputs, start_index=0):
         rotary_dim = ops.shape(inputs)[feature_axis]
         inverse_freq = self._get_inverse_freq(rotary_dim)
 
-        seq_len = ops.shape(inputs)[sequence_axis]
-        tensor = ops.arange(seq_len, dtype="float32") + start_index
+        if positions is not None:
+            # Use `positions` array instead of `start_index`
+            tensor = ops.cast(positions, dtype="float32")
+        else:
+            seq_len = ops.shape(inputs)[sequence_axis]
+            tensor = ops.arange(seq_len, dtype="float32") + start_index
```

Comment:
Let's move this branch into it's own sub function?

That might make it easier to try overriding.

```python
if positions is None:
    positions = self._compute_positions(inputs, start_index=0)
else:
    positions = ops.cast(positions, dtype="float32")
```

Reply:
Done.

Reply:
I would actually keep the `self._compute_positions` over-ridable method for a somewhat hacky reason. Since we don't have great "layer wrappinig" or generation mutation functionality right now, this allows for a patch on the layer itself that would control this computation.

Reply:
Done!